clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3405,6 +3406,105 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
   }
 }
 
+// Figure out the sub-group for the this function.  First we check the
+// attributes, then the global settings.
+static std::pair<LangOptions::SubGroupSizeType, int64_t>
+CalcEffectiveSubGroup(ASTContext &Ctx, const LangOptions &LO,
+                      FunctionDecl *FD) {
+  if (const auto *A = FD->getAttr<IntelReqdSubGroupSizeAttr>()) {
+    int64_t Val = getIntExprValue(A->getValue(), Ctx);
```

Comment:
Hrmmm, my spidey senses are tingling when I look at the definition of `getIntExprValue()` -- it looks like it's not set up to handle value dependence properly.

Reply:
I'm not familiar enough with the code to know, but because it's in Sema, my assumption is that we can devise a template to get us there. However, I'd say it's okay to wait until we find the problematic code path to address it because I can't quite come up with a test case in my head.